baker/Deployer.ts

```diff
@@ -0,0 +1,374 @@
+        else if (skipChecks) {
+            if (this.targetIsProd)
+                this.printAndExit(`Cannot skip checks when deploying to live`)
+            this.progressBar.tick({
+                name: "✅ finished checks because we skipped them",
+            })
+        } else {
+            await this.runAndTick(`yarn prettify:check`)
+            await this.runAndTick(`yarn build`)
+            await this.runAndTick(`jest`)
+        }
+
+        await this.writeHeadDotText()
+        await this.ensureTmpDirExistsOnServer()
+        await this.generateShellScriptsAndRunThemOnServer()
+
+        this.progressBar.tick({
+            name: `✅ 📡 finished everything`,
+        })
+        this.stream.replay()
+    }
+
+    // todo: the old deploy script would generete BASH on the fly and run it on the server. we should clean that up and remove these shell scripts.
+    private async generateShellScriptsAndRunThemOnServer() {
+        const { simpleGit } = this
+        const {
+            target,
+            userRunningTheDeploy,
+            owidGrapherRootDir,
+        } = this.options
+
+        const gitConfig = await simpleGit.listConfig()
+        const gitName = `${gitConfig.all["user.name"]}`
+        const gitEmail = `${gitConfig.all["user.email"]}`
+
+        const ROOT = "/home/owid"
+        const ROOT_TMP = `${ROOT}/tmp`
+        const SYNC_TARGET = `${ROOT_TMP}/${target}-${userRunningTheDeploy}`
+        const TMP_NEW = `${ROOT_TMP}/${target}-${userRunningTheDeploy}-tmp`
+        const FINAL_TARGET = `${ROOT}/${target}`
+
+        const scripts: any = {
+            file: makeScriptToDoFileStuff(ROOT, target, SYNC_TARGET, TMP_NEW),
+            yarn: makeScriptToDoYarnStuff(TMP_NEW),
+            adminServerStuff: makeScriptToDoQueueStuffDoFileStuffDoAdminServerStuff(
+                target,
+                ROOT_TMP,
+                TMP_NEW,
+                FINAL_TARGET
+            ),
+            bake: `pm2 stop ${target}-deploy-queue
+# Static build to update the public frontend code
+cd ${FINAL_TARGET}
+node itsJustJavascript/baker/bakeSite.js`,
+            deploy: makeScriptToDeployToNetlifyDoQueue(
+                target,
+                gitEmail,
+                gitName,
+                FINAL_TARGET
+            ),
+        }
+
+        Object.keys(scripts).forEach((name) => {
+            const fullName = `${name}${TEMP_DEPLOY_SCRIPT_SUFFIX}`
+            const localPath = `${owidGrapherRootDir}/${fullName}`
+            fs.writeFileSync(localPath, scripts[name], "utf8")
+            fs.chmodSync(localPath, "755")
+        })
+
+        await this.copyLocalRepoToServerTmpDirectory()
+
+        for await (const name of Object.keys(scripts)) {
+            await this.runAndStreamScriptOnRemoteServerViaSSH(
+                `${SYNC_TARGET}/${name}${TEMP_DEPLOY_SCRIPT_SUFFIX}`
+            )
+        }
+    }
+
+    printAndExit(message: string) {
+        // eslint-disable-next-line no-console
```

Comment:
Yeah I think we should get rid of that eslint rule, at least for server-side / node code...

Reply:
Hmm, I kind of like it? It has stopped me from leaving in console junk a number of times.